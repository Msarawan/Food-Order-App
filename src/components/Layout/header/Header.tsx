import * as React from 'react';
import { Fragment } from 'react';

import HeaderCartButton from '../headerCartButton/HeaderCartButton';
import mealsImage from '../../../assets/meals.jpg';
import classes from './Header.module.css';

interface IHeaderProps{
  onShowCart:()=>void
}


const Header: React.FC<IHeaderProps> = ({onShowCart}) => {
 return (
    <Fragment>
      <header className={classes.header}>
        <h1>Meals</h1>
        <HeaderCartButton  onClick={onShowCart}/>
      </header>
      <div className={classes['main-image']}>
        <img src={mealsImage} alt='A table full of delicious food!' />
      </div>
    </Fragment>
  );
};

export default Header;