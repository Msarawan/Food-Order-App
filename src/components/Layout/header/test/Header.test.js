import Header from '../../header/Header';
import React from "react";
import { mount } from "enzyme";

describe("Food Order app testing", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Header onClick='{showCartHandler}'/>);
  });

  test("render the title of the app", () => {
    expect(wrapper.find("h1").text()).toContain("Meals");
  });

  test("render state in the cart", () => {
    expect(wrapper.props().onClick).toEqual('{showCartHandler}');
    console.log(wrapper.find('Header').props())
  });

describe('Toggle Component', () =>{
  test('state of the cart to be true',()=>{
    const test = true;
    expect(test).toBe(true);
  })

  test('renders a snapshot', () => { 
    expect(wrapper).toMatchSnapshot();
  });
}) ;
});