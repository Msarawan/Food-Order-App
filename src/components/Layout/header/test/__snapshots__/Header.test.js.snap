// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Order app testing render state in the cart 1`] = `
<Header
  onClick="{showCartHandler}"
>
  <header
    className="header"
  >
    <h1>
      Meals
    </h1>
    <MockedHeaderCartButton />
  </header>
  <div
    className="main-image"
  >
    <img
      alt="A table full of delicious food!"
      src=""
    />
  </div>
</Header>
`;
